File: app/src/main/java/vn/tiki/sample/entity/Paging.java
package vn.tiki.sample.entity;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class Paging implements vn.tiki.collectionview.Paging {

  @AutoValue.Builder
  public static abstract class Builder {

    public abstract Paging make();

    public abstract Builder currentPage(int currentPage);

    public abstract Builder lastPage(int lastPage);

    public abstract Builder total(int total);
  }

  public static Builder builder() {
    return new AutoValue_Paging.Builder();
  }
}

File: sample/src/main/java/vn/tiki/sample/entity/Paging.kt
package vn.tiki.sample.entity

import paperparcel.PaperParcel
import paperparcel.PaperParcelable
import vn.tiki.collectionview.Paging

@PaperParcel
data class Paging(private val currentPage: Int = 1, private val lastPage: Int, private val total: Int = 0) : PaperParcelable, Paging {
  override fun currentPage() = currentPage

  override fun lastPage() = lastPage

  override fun total() = total

  companion object {
    @JvmField
    val CREATOR = PaperParcelPaging.CREATOR
  }
}